### PR TITLE
Run all the jupyter notebooks from the example directory as functional tests in CI

### PR DESCRIPTION
**Pull request by AndrewSazonov**

### CODE REVIEW

File: .github/workflows/test-ipynb.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+jobs:
+  test:
+    timeout-minutes: 30
+
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest]
+        python-version: ['3.10', '3.11', '3.12']
```
Comment (rozyczko) [inline .github/workflows/test-ipynb.yml:19]: I think we should concentrate on a single python version (3.12?) but include more OS-es. This is especially important for assuring file paths are properly set on Windows/Linux

File: .github/workflows/test-ipynb.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
+name: Testing Jupyter Notebooks
+
+on:
+  push:
+    branches:
```
Comment (rozyczko) [inline .github/workflows/test-ipynb.yml:5]: As discussed, this should also include `on-dispatch` case

File: easydiffraction/Job.py [hunk 3]
```diff
@@ -893,11 +890,14 @@ def print_free_parameters(self):
                 parameters['names'].append(parameter.display_name)
                 parameters['values'].append(parameter.raw_value)
                 parameters['errors'].append(parameter.error)
-                parameters['units'].append(f'{parameter.unit:~H}')
+                parameters['units'].append(f'{parameter.unit:~P}')
```
Comment (rozyczko) [inline easydiffraction/Job.py:893]: at some point, we need to refactor out the plotting and printing methods from `Job.py`. This should be easily possible. (Not now, let's wait until you are done with the main functionality)

File: easydiffraction/Job.py [hunk 4]
```diff
@@ -59,6 +60,8 @@
 except ImportError:
     print("pandas not installed")
 
+if 'JPY_PARENT_PID' in os.environ:
```
Comment (rozyczko) [inline easydiffraction/Job.py:63]: It would probably be good to add a comment that this means 'if the job is run from the notebook', as it isn't obvious

**Review by rozyczko — APPROVED** [review]
Looks good. Minor issues raised for discussion

File: .github/workflows/test-ipynb.yml (hunk 1 above)
Comment (AndrewSazonov) [reply .github/workflows/test-ipynb.yml:17]: Then let's have a full matrix with 3 python versions and 3 os versions to make sure everything is ok for all major combinations. Added to workflow config.

File: .github/workflows/test-ipynb.yml (hunk 2 above)
Comment (AndrewSazonov) [reply .github/workflows/test-ipynb.yml]: Replaced the `push` with the `workflow_dispatch`.

File: easydiffraction/Job.py (hunk 3 above)
Comment (AndrewSazonov) [reply easydiffraction/Job.py:901]: Completely agree with that.

File: easydiffraction/Job.py (hunk 4 above)
Comment (AndrewSazonov) [reply easydiffraction/Job.py]: Added a method to determine if code is running in Jupyter notebook.

**Pull request merged**